src/main/java/seedu/address/nusmods/ModuleInfo.java

```diff
@@ -28,6 +33,14 @@ public String getModuleCredit() {
         return moduleCredit;
     }
 
+    public String getSemesters() {
+        StringBuilder semesterResult = new StringBuilder();
+        for (Integer sem: semesters) {
+            semesterResult.append(sem).append(", ");
+        }
+        return semesterResult.substring(0, semesterResult.length() - 2);
```

Comment:
This will throw an error if there are no semesters, which makes `semesterResult` empty, cos then we'll be calling `.substring(0, -2)`. We can avoid dealing with extra commas and spaces by not adding them until they're needed: 
```suggestion
    public String getSemesters() {
        return semesters.stream().map(Object::toString).collect(Collectors.joining(", "));
    }    

```

Reply:
Wow back at it with the streams again 💯